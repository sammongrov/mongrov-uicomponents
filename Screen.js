import React, { Component } from 'react';
import { View, SafeAreaView } from 'react-native';
import PropTypes from 'prop-types';
import { styles } from 'react-native-theme';
import { Colors } from '@ui/theme_default';
import LinearGradient from 'react-native-linear-gradient';

export default class Screen extends Component {
  render() {
    const { children, safeBgColors, safeGradLocations, onLayout } = this.props;
    return (
      <LinearGradient colors={safeBgColors} style={{ flex: 1 }} locations={safeGradLocations}>
        <SafeAreaView style={{ flex: 1 }}>
          <View style={styles.authScreenContainer} onLayout={onLayout}>
            {children}
          </View>
        </SafeAreaView>
      </LinearGradient>
    );
  }
}

Screen.propTypes = {
  children: PropTypes.node.isRequired,
  safeBgColors: PropTypes.array,
  safeGradLocations: PropTypes.array,
  onLayout: PropTypes.func,
};

Screen.defaultProps = {
  safeBgColors: [Colors.BG_DEFAULT_SAFECOLOR_TOP, Colors.BG_DEFAULT_SAFECOLOR_BOTTOM],
  safeGradLocations: [0, 1],
  onLayout: () => {},
};
