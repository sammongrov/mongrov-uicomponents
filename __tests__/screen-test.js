import React from 'react';
import renderer from 'react-test-renderer';
// import { Alert } from 'react-native';
import Screen from '../Screen';

// const onHelpPress = () => Alert.alert('Help');

it('Screen renders correctly', () => {
  const tree = renderer.create(<Screen> hello </Screen>).toJSON();

  expect(tree).toMatchSnapshot();
});

test('name', () => {
  // array
  expect(Array.isArray(['value'])).toBe(true);
});
