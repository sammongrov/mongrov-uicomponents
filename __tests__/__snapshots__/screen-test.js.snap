// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Screen renders correctly 1`] = `
<BVLinearGradient
  colors={
    Array [
      4293914607,
      4293914607,
    ]
  }
  locations={
    Array [
      0,
      1,
    ]
  }
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTSafeAreaView
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      onLayout={[Function]}
    >
       hello 
    </View>
  </RCTSafeAreaView>
</BVLinearGradient>
`;
